app/controllers/spree/admin/adjustments_controller.rb

```diff
@@ -1,12 +1,23 @@
 module Spree
   module Admin
-    AdjustmentsController.class_eval do
+    class AdjustmentsController < ResourceController
+      belongs_to 'spree/order', find_by: :number
+      destroy.after :reload_order
```

Comment:
What's the reason for the `destroy.after` syntax here instead of `after_destroy`?

Reply:
interesting, I think it's the resource_controller callbacks extension... it's spree code and it's still in master:
https://github.com/spree/spree/blob/597cd1578705b49de44525efefdc7805a40cfa6b/backend/app/controllers/spree/admin/adjustments_controller.rb#L6

https://github.com/spree/spree/blob/master/backend/lib/spree/backend/action_callbacks.rb

As I understand it, this is used in resource_controller to setup callbacks, for example, after destroy, that are executed after the object is updated but before rendering:
https://github.com/spree/spree/blob/597cd1578705b49de44525efefdc7805a40cfa6b/backend/app/controllers/spree/admin/resource_controller.rb#L79

Reply:
basically, controller level callbacks. IMO Rails at it's worse where callback logic "learned" in models is extrapolated to controllers.

Reply:
Ugh. Yeah, it's not pretty...